parquet-protobuf/src/main/java/org/apache/parquet/proto/ProtoMessageConverter.java

```diff
@@ -80,12 +80,18 @@
       Descriptors.FieldDescriptor protoField = protoDescriptor.findFieldByName(parquetField.getName());
 
       if (protoField == null) {
+        // Skip extra list/key_value layer for spec-compliant, 3-level structures,
+        // when the proto schema doesn't have a field with this name.
+        if (specCompliant &&
+          ("key_value".equals(parquetField.getName()) || "list".equals(parquetField.getName()))) {
+          continue;
+        }
         String description = "Scheme mismatch \n\"" + parquetField + "\"" +
```

Comment:
This looks like a hacky way to fix the problem, it might be hard to maintain for later.